### ISSUE TITLE
multiple-instance-environment example doesn't load env variables

### ISSUE THREAD

**Issue by pbrandone** [danielcondemarin/serverless-next.js#228]
**Describe the bug**
The multiple environment example isn't loading the env variables to populate the `domain` key in the yml.

When logging the output from `await template({ template: 'serverless.yml' })` I'm getting 
```
{"AppNameHere":{"appUrl":"cloudfrontURL,"bucketName":"someRandomBucketName"}}
```

However, the environment variable `domain` is correctly set and available when the `template` function is executed, **so it seems the domain is effectively being ignored**.

**Expected behavior**
env variables to be used by serverless.yml

---

This is my first time using serverless so if this falls out of this package's scope please let me know.

**Comment by pbrandone** [comment]
My serverless.yml:
```yaml
EnzoWeb:
  component: serverless-next.js
  inputs:
    domain: ${env.domain}
```

and serverless.js
```js
const { Component } = require('@serverless/core')

/**
 * Will deploy the serverless component
 */
class Deploy extends Component {
  async default(inputs = {}) {
    const { stage } = inputs
    if (stage === 'staging' || stage === 'production') {
      // Will load env-${stage} as environment variables
      require('dotenv').config({ path: `${__dirname}/env-${stage}` })

      const template = await this.load('@serverless/template', stage)
      const output = await template({ template: 'serverless.yml' })

      return output
    } else {
      this.context.log(
        'No environment defined... Choices are staging and production'
      )
    }
  }

  // Remove will allow the stage to be set on the command line
  // run like so: serverless remove --stage=staging
  async remove(inputs = {}) {
    const { stage } = inputs
    if (stage === 'staging' || stage === 'production') {
      const template = await this.load('@serverless/template', stage)
      const output = await template.remove()
      return output
    }
  }
}

module.exports = Deploy
```

I have both `env-production` and `env-staging` files and my package.json scripts are setting stage with `serverless --stage=`

**Comment by barrysteyn** [comment]
Look at the code: The component is not returning the domain, so it can't be in the output: https://github.com/danielcondemarin/serverless-next.js/blob/master/packages/serverless-nextjs-component/serverless.js#L384

**Comment by pbrandone** [comment]
@barrysteyn that's true, but still doesn't answer the question.

Looking at that file it seems that inputs.domain IS NOT an instance of Array (it should probably be a string if it comes from env variables).
https://github.com/danielcondemarin/serverless-next.js/blob/master/packages/serverless-nextjs-component/serverless.js#L371

**Comment by pbrandone** [comment]
fixed by using
```yaml
domain:
      - ${env.subdomain}
      - mydomain.com
``` 
in the .yml

thanks for the help @barrysteyn

**Comment by pbrandone** [comment]
@barrysteyn let me know if you guys need a PR using this approach for the example. maybe two separate env variables `subdomain` and `domain`?

**Comment by barrysteyn** [comment]
I understand now, but I don't think this is a generic solution. Can you reopen this issue. Instead, I think we need a way to either represent the a YML array in a env variable, or we should alter the component such that if any part of the domain array is null (or undefined), it will get skipped. In that case, one can just define env vars, and leave subdomain empty for prod.

But first prize would be to represent an array in an env var.

**Comment by barrysteyn** [comment]
@pbrandone I don't have any domains on Route 53, so I can't test things. Would be grateful if you could test. Looking at https://discuss.elastic.co/t/array-support-for-yml-environment-values/76657
I think we should change the domain in the env files. Can you try this instead:

```bash
domain='dev, mydomain.com'
```

And if the above is not interpreted as an array, can you try this instead:

```bash
domain='-dev, -mydomain.com'
```

**Comment by pbrandone** [comment]
@barrysteyn of course. My project runs a custom server.js to proxy requests to /graphql to the backend so it seems that I might go with a beanstalk with EC2 approach for deployments.

I can however leave this branch as is to test this issue as it's working very well.

so far both `domain='dev, mydomain.com'` and `domain='-dev, -mydomain.com'` didn't work, any other suggestions?

**Comment by barrysteyn** [comment]
Hi, I don't have any other suggestions for now. I need to do some research regarding representing arrays as env variables for yml. For that, i will need to dig into the @sereverless/template component to see what yml parser it is using.

Keep this issue open for now,  I will communicate here when I get a better answer.

**Comment by pbrandone** [comment]
ok. thanks for the help.

for all it's worth, you could:
```js
const domain = typeof inputs.domain === 'string' ? JSON.parse(inputs.domain) : inputs.domain
if (domain instanceof Array) {
// ...etc
```
here https://github.com/danielcondemarin/serverless-next.js/blob/master/packages/serverless-nextjs-component/serverless.js#L371

but honestly I don't know if it's safe

**Comment by pbrandone** [comment]
@barrysteyn unfortunately it doesn't.

```
appUrl:     https://undefined.mydomain.com
```

**Comment by pbrandone** [comment]
I guess it gets @serverless/template replaces `undefined` with `'undefined'`

**Comment by pbrandone** [comment]
However, when using www as subdomain everything worked flawlessly.

I got `https://mydomain.com` as `appUrl` and in Route53 I have both pointing at the correct cloudfront instance

**Comment by barrysteyn** [comment]
He he - unfortunately, I think www as a subdomain is working because you have some rule (probably in the DNS) that is forwarding making it go to the place you want.

The way forward is to actually change the code of the component. Keep the ticket open, I will try change the code and then ask you to test.

Thanks so much for testing this.

**Comment by pbrandone** [comment]
sure thing.

I actually checked the DNS configuration and removed all the records I added before for my beanstalk configuration.

so it seems that @serverless/domain created both www and non-www automatically

**Comment by barrysteyn** [comment]
@pbrandone You are correct: using `www` is the correct thing and it will work like magic. See https://github.com/serverless-components/domain/blob/master/serverless.js#L213

However, I am going to make a PR that will be more intuitive for users.

**Comment by barrysteyn** [comment]
@pbrandone Can you test your stuff on my PR #231 
Tell me if this works.

**Comment by pbrandone** [comment]
@barrysteyn it's working :)

**Comment by barrysteyn** [comment]
Hi, I just needed to push another version of the PR since the tests were failing. I also took it upon myself to make the code cleaner. While I can't think of any reason this version would fail, can I ask you to do just one last test? @danielcondemarin will probably merge soon. Thank you very much for  bringing this matter up and helping out.

**Comment by pbrandone** [comment]
Hi @barrysteyn.

Just tested it again and it still works 👍

**Comment by pbrandone** [comment]
Closed by #231
